lib/iex/lib/iex/evaluator.ex

```diff
@@ -97,25 +97,33 @@ defmodule IEx.Evaluator do
     end
   end
 
-  defp do_eval(@break_trigger, state=%IEx.State{cache: ''}) do
+  defp do_eval(@break_trigger, state=%IEx.State{cache: ''}, _) do
     state
   end
 
-  defp do_eval(@break_trigger, state) do
+  defp do_eval(@break_trigger, state, _) do
     :elixir_errors.parse_error(state.counter, "iex", "incomplete expression", "")
   end
 
-  defp do_eval(latest_input, state) do
+  @history_key :__history_proc__
+
+  defp do_eval(latest_input, state, history) do
     code = state.cache ++ latest_input
     line = state.counter
-    handle_eval(Code.string_to_quoted(code, [line: line, file: "iex"]), code, line, state)
+
+    # Put the history pid into process dictionary so that IEx.Helpers can
+    # access it
+    Process.put(@history_key, history)
```

Comment:
While this is better it still suffers of the same issue that we are relying on the user process pdict. One alternative is to add a [whereis function similar to the server one](https://github.com/lexmag/elixir/blob/iex-history/lib/iex/lib/iex/server.ex#L14-L32). Since the server has the evaluator in its pdict, we can then look it up and find the history in the evaluator pdict.

So we are still using pdicts but not in the user process. What do you think?